libs/remote_storage/src/s3_bucket.rs

```diff
@@ -537,16 +518,20 @@ impl RemoteStorage for S3Bucket {
 
             match resp {
                 Ok(resp) => {
+                    // our ability to delete something does not necessarily correlate to us being able to
+                    // witness a success, but doing this only on "successes" or "not failures" might leave out
```

Comment:
This isn't easy to read. I am saying this who often likes to write extremely long sentences. Maybe turn this paragraph into two sentences?

Reply:
Maybe also worth it to make it a separate function to have one comment in one place instead of two? Also we probably can deprecate singular `delete_object` api because its fully covered with `delete_objects`

Reply:
3f2e2b56b implements delete with delete_objects, but does nothing larger.

I just removed the comment in b21ed7712. I did think about it and retries and then wrote the comment, but perhaps is not useful at all.

Reply:
I'm okay with having a comment here. It should just be easier to read :).

Reply:
Well, it'd be repeating the existing delete comments about s3 allowing multiple deletes, so I am thinking this is not needed after all.